### ISSUE TITLE
GM_addStyle

### ISSUE THREAD

**Issue by TimidScript** [greasemonkey/greasemonkey#2216]
Is it possible to allow GM_addStyle to behave a bit like Stylish, in the sense it overrides any styles present in the document. It has precedence.

If the answer is yes to the first, then would you also consider allowing two more parameters to add "id" and maybe even the ability to choose to add it in an iframe document. 

The id allows replacement of style. Currently I use my own library (`TSL.addStyle(id,css,doc)`) to do this, but it lacks precedence over page styles, and I have to mess around with selectors and important to override them.

Thanks

**Comment by arantius** [comment]
This is the sort of feature that should be implemented in a script itself, perhaps a reusable `@require` library.  I would not plan on adding it to Greasemonkey.

**Comment by TimidScript** [comment]
I do have the <a href="https://openuserjs.org/libs/TimidScript/TSL_-_Generic">library</a> but it does not provide override feature. This is done on a lower level by an extension. 

I was wondering if you can implement the precedence override that stylish has. Stylish somehow overrides every style by force. No amount of scripting from a userscript can override Stylish styles. One cannot even alter them using FireFox Web Developer Tools.  I was wondering if you can provide that level of precedence. 

If you add the override feature, it would also be nice to add the other two items I mentioned (id and doc). If no for all well that's that. ^_^

**Comment by pyhedgehog** [comment]
What do you mean by override? Adding `!important` for each style attribute?
I'm using Stylish, but have to add it everywhere...

**Comment by TimidScript** [comment]
> What do you mean by override? Adding !important for each style attribute?

It seems I am making blunder after blunder last few days. I have no clue what I did last time. But I was just going to prove you wrong and guess I was wrong. Last week I couldn't override the colour scheme of stylish from a userscript or web developer tools, no matter what I did. Spent hours and kept failing. Took me few minutes now to prove I must have been high.

Only reason I suggested it, is because of mistaken belief. The suggestion is more than useless.
